File: sdk/python/legend_ai/resources/charts.py
"""
Chart generation resource
"""

from typing import Optional, List, TYPE_CHECKING
from ..models import ChartResult

if TYPE_CHECKING:
    from ..client import LegendAI, AsyncLegendAI


class ChartsResource:
    """Synchronous chart generation resource"""

    def __init__(self, client: "LegendAI"):
        self._client = client

    def generate(
        self,
        ticker: str,
        interval: str = "1day",
        entry: Optional[float] = None,
        stop: Optional[float] = None,
        target: Optional[float] = None,
        indicators: Optional[List[str]] = None,
    ) -> ChartResult:
        """
        Generate a professional chart

        Args:
            ticker: Stock ticker symbol
            interval: Time interval
            entry: Entry price to mark on chart
            stop: Stop loss price to mark
            target: Target price to mark
            indicators: List of indicators to add

        Returns:
            ChartResult with chart URL
        """
        payload = {
            "ticker": ticker,
            "interval": interval,
        }
        if entry is not None:
            payload["entry"] = entry
        if stop is not None:
            payload["stop"] = stop
        if target is not None:
            payload["target"] = target
        if indicators:
            payload["indicators"] = indicators

        response = self._client.request("POST", "/api/charts/generate", json=payload)
        return ChartResult.from_dict(response)


class AsyncChartsResource:
    """Asynchronous chart generation resource"""

    def __init__(self, client: "AsyncLegendAI"):
        self._client = client

    async def generate(
        self,
        ticker: str,
        interval: str = "1day",
        entry: Optional[float] = None,
        stop: Optional[float] = None,
        target: Optional[float] = None,
        indicators: Optional[List[str]] = None,
    ) -> ChartResult:
        """Generate a professional chart (async)"""
        payload = {
            "ticker": ticker,
            "interval": interval,
        }
        if entry is not None:
            payload["entry"] = entry
        if stop is not None:
            payload["stop"] = stop
        if target is not None:
            payload["target"] = target
        if indicators:
            payload["indicators"] = indicators

        response = await self._client.request("POST", "/api/charts/generate", json=payload)
        return ChartResult.from_dict(response)

File: sdk/javascript/src/resources/charts.ts
/**
 * Chart generation resource
 */

import type { AxiosInstance } from 'axios';
import type { ChartGenerateRequest, ChartGenerateResponse, ChartResult } from '../types';

export class ChartsResource {
  constructor(private client: AxiosInstance) {}

  /**
   * Generate a professional chart
   *
   * @param ticker - Stock ticker symbol
   * @param options - Chart options
   * @returns Chart generation result
   */
  async generate(
    ticker: string,
    options: Omit<ChartGenerateRequest, 'ticker'> = {}
  ): Promise<ChartResult> {
    const response = await this.client.post<ChartGenerateResponse>('/api/charts/generate', {
      ticker,
      ...options,
    });
    return {
      chart_url: response.data.chart_url,
      cached: response.data.cached,
      processing_time: response.data.processing_time,
    };
  }
}
